### ISSUE TITLE
Layer won't deploy: "Layer conversion failed" error

### ISSUE THREAD

**Issue by dan-mba** [aws/aws-sam-cli#2344]
### Description

Deploying my application fails on layer creation with a file permission error.
I tried to `chmod -R 755` the layer build directory in git bash, but it had no effect.

### Steps to reproduce

1. clone my [repo](https://github.com/dan-mba/aws-sam-rekognition)
1. sam build --use-container
1. sam deploy --guided

### Observed result

CREATE_FAILED
AWS::Lambda::LayerVersion
OpencvLayer641c5be385
Layer conversion failed: Some files do not have read permissions;
(Service: AWSLambdaInternal; Status Code: 400; Error Code: InvalidParameterValueException;
Request ID: b212fad5-ef52-4386-b1de-d391533f9c68)

### Expected result

Deployment to complete

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: Windows 8.1
2. `sam --version`: `SAM CLI, version 1.7.0`

**Comment by dan-mba** [comment]
I was able to get my code working by moving all of my dependencies into my Lambda.
See the dev branch of my repo.

But with my bundle being almost 190MB with 99% being dependencies, 
this is not a good way to develop quickly having to constantly re-uploading my dependencies.

**Comment by codestart** [comment]
I am getting this exact error with my Node.js app. Thankfully my bundle is only 13.8MB but still, not ideal. This is what I am doing as a workaround too.
I have the same code as I had a few weeks back (that worked) and my code runs fine when run locally. The only change (I haven't rolled-back) is the change to SAM CLI 1.7.0... unless I'm missing something :(

**Comment by codestart** [comment]
@dan-mba I've uninstalled v1.7.0 and installed v1.6.2 and it runs fine 👍

**Comment by singledigit** [comment]
Hey @dan-mba, is this layer being managed in the same template? Or is it an external layer? Can you post a copy of the SAM template?

**Comment by dan-mba** [comment]
@singledigit The template is in my [repo here](https://github.com/dan-mba/aws-sam-rekognition).

The source for the layer is only the requirements.txt for the dependencies. They contain platform specific modules, which is why I have to build with container.

**Comment by jholladay10** [comment]
FYI, I was having this exact problem (python).  I downgraded to SAM 1.6.2, re-deployed my layer stack, and still had the problem.  Then I re-deployed my function stack using 1.6.2 and it worked.  Also note that I'm having trouble getting my layer to be assigned via the SAM function template.  My layer is in a different stack.  So my scenario at the moment is:
1. Deploy lambda layer stack
2. Deploy lambda function stack
3. Wire it up in the console.  This is where I was encountering the error in this thread.

Anyway, bottom line: if you're having this problem after reverting to 1.6.2, make sure you re-deploy the layers and functions.

**Comment by singledigit** [comment]
This is a known regression in v1.7.0 and the team is currently working on a fix. Thanks for reporting!!

**Comment by julianwood** [comment]
Had the same issue. I tried multiple SAM versions, from 1.6.0 to 1.8.0 on both Window and Linux (WSL2), looking at same files.
I then recreated a new folder structure and copied the same files in again, did a chmod -R +r *.

**Comment by awsjeffg** [comment]
@dan-mba closing this as SAM cli version 1.9.0 is now out with a fix for this issue. Please re-open this issue if you update and see the same problem, or file a new issue if you find other issues. Thanks again for reporting this!
